ohapi/api.py

```diff
@@ -190,15 +182,41 @@ def message(subject, message, access_token, all_members=False,
         base_url, '/api/direct-sharing/project/message/?{}'.format(
             urlparse.urlencode({'access_token': access_token})))
     if not(all_members) and not(project_member_ids):
-        return requests.post(url, data={'subject': subject,
-                                        'message': message})
+        response = requests.post(url, data={'subject': subject,
+                                            'message': message})
+        handle_error(response)
+        return response
     elif all_members and project_member_ids:
         raise ValueError(
             "One (and only one) of the following must be specified: "
             "project_members_id or all_members is set to True.")
     else:
-        return requests.post(url, data={
-                             'all_members': all_members,
-                             'project_member_ids': project_member_ids,
-                             'subject': subject,
-                             'message': message})
+        r = requests.post(url, data={'all_members': all_members,
+                                     'project_member_ids': project_member_ids,
+                                     'subject': subject,
+                                     'message': message})
+        handle_error(r)
+        return r
+
+
+def handle_error(r):
+    code = r.status_code
+    if code in [200, 201, 202, 204]:
+        return
+    else:
+        info = 'API response status code {}'.format(code)
+        if 'detail' in r.json():
+            info = info + ": {}".format(r.json()['detail'])
+        raise Exception(info)
+        return
+
+
+def process_info(remote_file_info, filesize, target_filepath):
+    response = requests.get(remote_file_info['download_url'], stream=True)
+    remote_size = int(response.headers['Content-Length'])
+    if remote_size == filesize:
+        logging.info('Skipping {}, remote exists with matching name and '
+                     'file size'.format(target_filepath))
+        raise ValueError("Remote exist with matching name and file size")
```

Comment:
Will this ever return something if you raise the error?

Reply:
Actually it was in the original function added in a previous PR by Tarannum, so I kept it as it is. Had a doubt though. Nevertheless, correcting it. :)